### PR TITLE
Add Header component

### PR DESCRIPTION
**Pull request by augusto-jm-amaral**
closes: #18

### CODE REVIEW

File: webapp/src/components/Header/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import CloseIcon from 'react-icons/lib/io/android-close'
+import BackIcon from 'react-icons/lib/io/android-arrow-back'
+import {
+  string,
+  func,
+} from 'prop-types'
```
Comment (thiagommedeiros) [inline webapp/src/components/Header/index.js]: Acho que podemos convencionar que o import `prop-types` esteja sempre na sequência do import do react, faz sentido pra vcs?

Comment (vcapretz) [reply webapp/src/components/Header/index.js]: 👍

**Comment by felquis** [comment]
Por favor, tire um print dessa mensagem e mostre para o Henrique de segurança :)

Nos vemos de novo no futuro ;)

Abs!

**Comment by thiagommedeiros** [comment]
@felquis Seria legal vc nos contar o motivo, vc identificou algum problema no código?

**Comment by GitCop** [comment]
Os seguintes commits de seu Pull Request precisam de ajustes no título:

 * Commit: 8441d6715a7b434e367f8039bdd7798bfd5b166b
  * Utilize o padrão `escopo: ação descrição`, sempre em minúsculo e inglês.  
  
 - O **escopo** não deve conter espaços. Se a palavra for composta, separe por traços  
    - A **ação** deve ser **add**, **remove**, **rename**, **refactor**, **move**,  **bump**, **update** ou **release**, **fix**  
    - A **descrição** deve conter uma rápida descrição da alteração

O guia completo de estilo de commits está disponível em https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by thiagommedeiros** [comment]
@augusto-jm-amaral I think we can squash the 'commits'. Two 'commits' is sufficient to this task: 
- `header: add header component`
- `stories: add header story`.

**Comment by thiagommedeiros** [comment]
@augusto-jm-amaral The logo images are not the same as the layout, and the image sizes are too large. You can request to the UI Team the correct images.

File: webapp/config/webpack.config.storybook.js [hunk 2]
```diff
@@ -67,7 +67,7 @@ module.exports = {
         ],
       },
       {
-        test: /\.(woff|woff2|eot|ttf|svg)$/,
+        test: /\.(woff|woff2|eot|ttf|svg|png)$/,
```
Comment (thiagommedeiros) [inline webapp/config/webpack.config.storybook.js]: This commit message could be `storybook: add png rule to webpack config`.

**Comment by augusto-jm-amaral** [comment]
I adjusted the commits, I'll arrange the logo, thanks @thiagommedeiros .

File: webapp/src/components/Header/index.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  logoSrc,
+  logoAlt,
+  onPrev,
+  onClose,
+}) => (
+  <header className={className}>
+    <Grid>
+      <Row className={style.row}>
+        <Col
+          className={style.buttonContainer}
```
Comment (thiagommedeiros) [inline webapp/src/components/Header/index.js]: We do not need this class anymore.

File: webapp/src/components/Header/index.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            base={base}
+            fill="clean"
+            onClick={onPrev}
+            relevance="normal"
+          >
+            <BackIcon />
+          </Button>
+        </Col>
+        <Col
+          className={style.logoContainer}
```
Comment (thiagommedeiros) [inline webapp/src/components/Header/index.js]: Same here.

**Comment by vcapretz** [comment]
I noticed the component is not vertically aligned and when debugging I saw a margin leaking, as it shows in the image bellow (I may be wrong but it's a bad practice and it should not happen) 

screen shot 2017-12-08 at 10 01 06

File: webapp/src/components/Header/Header.test.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+import Header from './index'
+
+describe('Header', () => {
+  it('should mount', () => {
+    const component = mount(
+      <Header />
+    )
+
+    expect(component.find('img').first()).toBeDefined()
```
Comment (vcapretz) [inline webapp/src/components/Header/Header.test.js]: do not use first and use `toHaveLength` = the expected value it should have (applies for the cases below as well)

File: webapp/src/components/Header/Header.test.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    expect(component.find('button').length).toBe(2)
```
Comment (vcapretz) [inline webapp/src/components/Header/Header.test.js]: `expect(component.find('button').length).toBe(2)` -> 
`expect(component.find('button')).toHaveLength(2)`

File: webapp/src/components/Header/index.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  logoSrc,
+  logoAlt,
+  onPrev,
+  onClose,
+}) => (
+  <header className={className}>
+    <Grid>
+      <Row>
+        <Col
+          tv="4"
```
Comment (vcapretz) [inline webapp/src/components/Header/index.js]: these should be integers (use `const` to define it and let it be easier for us to track and modify later, as in https://github.com/mundipagg/mercurio-checkout/pull/40/commits/8cb0f73a19a6fd3c7c9398fc09b0cc7b4e5f016c) 

screen shot 2017-12-08 at 10 19 46

File: webapp/src/components/Header/properties.css [hunk 8]
```diff
@@ -0,0 +1,4 @@
+:root {
+  --logo-height: 50px;
```
Comment (vcapretz) [inline webapp/src/components/Header/properties.css]: when defining a fixed logo height and allowing users to set images of their own companies won't break the layout or load the image with different sizes? this way the user would have to override the img properties all the time (I really don't know how to procede with this, I set the image height as 100% and worked fine, but maybe we would need to define a few patterns for new images)

Comment (thiagommedeiros) [reply webapp/src/components/Header/properties.css]: I think we need to use a fixed value for width and height, and our documentation specifies the ideal value for both.

**Pull request merged**